tools/ddrgen/src/scanners/dwarf/AixDwarfParser.cpp

```diff
@@ -73,16 +73,25 @@ dwarf_finish(Dwarf_Debug dbg, Dwarf_Error *error)
 	/* Delete the built-in type DIEs */
 	if (NULL != _builtInTypeDie) {
 		deleteDie(_builtInTypeDie);
+		_builtInTypeDie = NULL;
 	}
 
 	/* Delete the undefined type reference DIE */
 	if (NULL != _undefinedRef) {
 		deleteDie(_undefinedRef);
+		_undefinedRef = NULL;
 	}
 
+	_lastDie = NULL;
 	Dwarf_CU_Context::_fileList.clear();
+	createdDies.clear();
```

Comment:
All of these collections should not be needed after dwarf_init() completes. At the end of dwarf_init, the main context's pointers should lead to all dwarf_die's. If the stabstring parsing is complete in dwarf_init, there is no reason to keep any of these collections until dwarf_finish().